config/settings/base.py

```diff
@@ -77,7 +77,8 @@
     "django.contrib.auth.middleware.AuthenticationMiddleware",
     "django.contrib.messages.middleware.MessageMiddleware",
     "django.middleware.clickjacking.XFrameOptionsMiddleware",
-    "adserver.middleware.XForwardedForMiddleware",
+    "adserver.middleware.ServerInfoMiddleware",
+    "adserver.middleware.CloudflareMiddleware",
```

Comment:
For GeoIP purposes, this is a fine default since it falls back to the GeoIP databases. For getting the true IP header, this isn't safe although the x-forwarded-for header is also unsafe to rely on unless deployed properly. Maybe these should be split into two middlewares. Regardless, we could just have an envvar like `ADSERVER_EXTRA_MIDDLEWARES` and then do something like:

```python
for middleware in ADSERVER_EXTRA_MIDDLEWARES:
    MIDDLEWARE.append(middleware)
```

Reply:
I refactored this a bit. There's now two settings:
- `ADSERVER_GEOIP_MIDDLEWARE`
- `ADSERVER_IPADDRESS_MIDDLEWARE`

These append middleware to the settings. There are provided middlewares for GeoIP databases, Cloudflare GeoIP, Cloudflare Real IP, and X-Forwarded-For.